src/js/draad-maps.js

```diff
@@ -9,839 +9,1078 @@ const mapEvent = new Event("draadMapsLoaded");
+								iconUrl: `${draadMapsConfig.pluginDir}/dist/images/marker-hover.png`
+							}),
+				active:
+					typeof node !== "undefined" &&
+					node.dataset.markerActive !== ""
+						? this.getLeafletIcon({
+								iconUrl: node.dataset.markerActive
+							})
+						: this.getLeafletIcon({
+								iconUrl: `${draadMapsConfig.pluginDir}/dist/images/marker-active.png`
+							})
+			};
+
+			feature._style = {
+				color:
+					typeof node !== "undefined" &&
+					node.dataset.shapeColor !== ""
+						? node.dataset.shapeColor
+						: this.colors.primary,
+				weight:
+					typeof node !== "undefined" &&
+					node.dataset.shapeWidth !== ""
+						? node.dataset.shapeWidth
+						: 4,
+				dashArray:
+					typeof node !== "undefined" &&
+					node.dataset.shapeStyle === "solid"
+						? "0, 0"
+						: "8, 8",
+				fillColor:
+					typeof node !== "undefined" &&
+					node.dataset.shapeColor !== ""
+						? node.dataset.shapeColor
+						: this.colors.primary,
+				fillOpacity: 0
+			};
+
+			// set popup
+			if (feature.feature.properties.titel) {
+				feature.bindPopup(feature.feature.properties.titel);
+			}
+
+			// set border
+			if (typeof feature.setStyle === "function") {
+				feature.setStyle(feature._style);
+				feature.options.alt = feature.feature.properties?.name;
+				feature.options.title = feature.feature.properties?.titel;
+				feature.selected = false;
+				this.dataHandler(feature);
+			}
+
+			// set icon
+			if (typeof feature.setIcon === "function") {
+				feature.setIcon(feature._styles.default);
+
+				feature.options.alt = feature.feature.properties?.name;
+				feature.options.title = feature.feature.properties?.titel;
+				feature.selected = false;
+				this.markerHandler(feature, null);
+			}
+		});
+
+		return layer;
+	};
+
+	/**
+	 * Event handler for GeoJSON data.
+	 *
+	 * @param {object} feature The GeoJSON feature.
+	 */
+	dataHandler = (feature) => {
+		const draad = this;
+		feature.on("click", (e) => {
+			this.dataSetState(feature, "active");
+			this.map.flyToBounds(feature.getBounds(), { padding: [0, 0] });
+		});
+
+		feature.on("popupclose", (e) => this.dataSetState(feature, "default"));
+	};
+
+	/**
+	 * Changes the state of a GeoJSON feature.
+	 *
+	 * @param {object} feature The GeoJSON feature.
+	 * @param {string} state The state of the feature.
+	 */
+	dataSetState = (feature, state) => {
+		switch (state) {
+			case "active":
+				const styleActive = feature._style;
+				styleActive.fillOpacity = 0.15;
+				feature.setStyle(styleActive);
+				break;
+			case "hover":
+				const styleHover = feature._style;
+				styleHover.fillOpacity = 0.15;
+				feature.setStyle(styleHover);
+				break;
+			case "focus":
+				const styleFocus = feature._style;
+				styleFocus.fillOpacity = 0.15;
+				feature.setStyle(styleFocus);
+				break;
+			default:
+				const styleDefault = feature._style;
+				styleDefault.fillOpacity = 0;
+				feature.setStyle(styleDefault);
+				break;
+		}
```

Comment:
Suggestie voor simpelere JS:
```suggestion
		const style = feature._style;
		switch (state) {
			case "active":
			case "hover":
			case "focus":
				style.fillOpacity = 0.15;
				break;
			default:
				style.fillOpacity = 0;
				break;
		}
		feature.setStyle(style);
```